File: src/thermostat.js
'use strict'

class Thermostat {

  constructor() {
    this.MINIMUM_TEMPERATURE = 10;
    this.MAXIMUM_TEMPERATURE = 25;
    this.PSM = true
    this.temperature = 20;
  }

  getCurrentTemperature() {
    return this.temperature;
  }

  up() {
    if(this.PSM === true && this.temperature >= this.MAXIMUM_TEMPERATURE) {
      return this.temperature = this.MAXIMUM_TEMPERATURE;
    }
    this.temperature += 1;
  }

  down() {
    if (this.temperature === this.MINIMUM_TEMPERATURE) {
      return;
    };
    this.temperature -= 1;
  }

  isPowerSavingModeOn() {
    return this.PSM === true;
  }

};

File: lib/thermostat.rb
require 'pg'

class Thermostat

  def self.save_to_DB(temperature:, psm:, city:)
    db_connect = PG.connect dbname: 'thermostat'
    db_connect.exec("TRUNCATE data_saved;")
    db_connect.exec("INSERT INTO data_saved (temperature, PSM, city) VALUES ('#{temperature}', '#{psm}', '#{city}');")
  end

  def self.list_from_DB
    db_connect = PG.connect dbname: 'thermostat'
    db_connect.exec("SELECT * FROM data_saved;")
  end

end
